cluster/swarm/cluster.go

```diff
@@ -330,6 +330,7 @@ func (c *Cluster) validatePendingEngine(engine *cluster.Engine) bool {
 	// set engine state to healthy, and start refresh loop
 	engine.ValidationComplete()
 	c.engines[engine.ID] = engine
+	engine.AfterEngineValidationComplete()
```

Comment:
I think this is redundant. If the call `engine.Connect` on line 295 succeeds, it already emits `engine_connect` event (See `ConnectWithClient`).